### ISSUE TITLE
Order not preserved while sending data

### ISSUE THREAD

**Issue by dummy26** [peers/peerjs#746]
This is a part of my code that I am using to transfer files.
```js
 const CHUNK_SIZE = 16 * 1024
 file = e.target.files[0]
 let i
 let idx = 0
 for (i = 0; i < file.size; i += CHUNK_SIZE) {
     connection.send({ chunk: file.slice(i, i + CHUNK_SIZE), idx })
     idx++
 }
```

When this data is received on the other user's end then I print the idx received and all of them are in the right order except the last two.
Example - 
0
1
2
.
.
.
207
**209
208**

This is happening for all the files but most of the files are opening without any problem while some of them are not.

I tried setting the ordered attribute to true but it still remained false -
```js
 connection = peer.connect(userId)
 connection.dataChannel.ordered = true
```

**Comment by Swanand01** [comment]
The same has been happening with me. The second to last chunk is always received before the last one.

**Comment by cohenstyle** [comment]
Yeah. It's not setting the DC to ordered unfortunately. Having the same issue.

**Comment by cohenstyle** [comment]
Thanks a lot!
I was using it with files though so I don't have a setup that can test it.

**Comment by brianismeta** [comment]
I was having an ordering issue but I thought it had to do with UDP... there are many data packets being transferred between two peers, and sometimes it gets out of order very quickly, or it gets out of order after a few hundred packets.  I forced communication through TURN and used TCP channel, and the issue was resolved, as far as I could tell.  I am assuming it was still a UDP issue but after reading this issue - is it possible it could have been this issue -- or does this issue only occur at the very "end" of the data channel communications (??)
